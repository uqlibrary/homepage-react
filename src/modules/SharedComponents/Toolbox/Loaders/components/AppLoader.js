import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@mui/material/CircularProgress';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import { styled } from '@mui/material/styles';

const StyledLoader = styled(Grid)(({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    ...(((theme.palette || {}).primary || {}).gradient || {}).diagonal,
    width: '100%',
    height: '100%',
    textAlign: 'center !important',

    '& .white': {
        color: ((theme.palette || {}).white || {}).main,
        fontWeight: (theme.typography || {}).fontWeightLight,
    },
    '& .spaceBetween': {
        margin: '16px 0',
    },
    '& .logo': {
        width: 200,
    },
}));

export class AppLoader extends React.Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        logoImage: PropTypes.string,
        logoText: PropTypes.string,
    };

    render() {
        const { title, logoImage, logoText } = this.props;
        return (
            <StyledLoader container spacing={0} direction="column" justifyContent="center" alignItems="center">
                <Grid item className={'spaceBetween'}>
                    <CircularProgress
                        size={80}
                        thickness={1}
                        className={'white'}
                        aria-label="Loading Library website"
                    />
                </Grid>
                <Grid item className={'spaceBetween'}>
                    {logoImage && <div className={`${logoImage} logo`} alt={logoText} />}
                </Grid>
                <Grid item className={'spaceBetween'}>
                    <Typography variant={'h6'} className={'white'}>
                        {title}
                    </Typography>
                </Grid>
            </StyledLoader>
        );
    }
}

export default AppLoader;
