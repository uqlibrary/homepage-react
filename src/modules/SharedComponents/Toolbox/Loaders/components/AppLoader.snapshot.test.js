import { AppLoader } from './AppLoader';
import AppLoaderWithStyles from './AppLoader';

function setup(testProps = {}) {
    const props = {
        ...testProps,
        classes: {
            appLoader: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                backgroundColor: 'red',
                width: '100%',
                height: '100%',
                textAlign: 'center !important',
            },
            white: {
                color: 'white',
                fontWeight: 200,
            },
            spaceBetween: {
                margin: '16px 0',
            },
        },
    };
    return getElement(AppLoader, props);
}

describe('Component AppLoader', () => {
    it('should render as expected', () => {
        // set initial props values
        const props = {
            title: 'Fez frontend',
            logoImage: 'http://image/image.svg',
            logoText: 'Fez logo',
        };
        const wrapper = setup({ ...props });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with styles', () => {
        const wrapper = getElement(AppLoaderWithStyles, { title: 'Test loader' });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
