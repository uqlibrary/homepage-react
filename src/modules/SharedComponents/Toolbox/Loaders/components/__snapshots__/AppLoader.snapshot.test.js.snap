// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AppLoader should render as expected 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-7fsne-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item spaceBetween css-13i4rnv-MuiGrid-root"
    >
      <span
        aria-label="Loading Library website"
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary white css-a0wk1a-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="21.5"
            stroke-width="1"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item spaceBetween css-13i4rnv-MuiGrid-root"
    >
      <div
        alt="Fez logo"
        class="http://image/image.svg logo"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item spaceBetween css-13i4rnv-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 white css-1uywuy6-MuiTypography-root"
      >
        Fez frontend
      </h6>
    </div>
  </div>
</div>
`;
